coreruleset.go

```diff
@@ -4,16 +4,65 @@ import (
 	"embed"
 	"io/fs"
 	"log"
+	"strings"
 )
 
+// subFS is the interface that contains the methods that are used by
+// fs.subFS
+type subFS interface {
+	Open(name string) (fs.File, error)
+	ReadDir(name string) ([]fs.DirEntry, error)
+	ReadFile(name string) ([]byte, error)
+	Glob(pattern string) ([]string, error)
+}
+
 //go:embed rules/*
 var wFS embed.FS
 var FS fs.FS
 
 func init() {
-	var err error
-	FS, err = fs.Sub(wFS, "rules")
+	rulesFS, err := fs.Sub(wFS, "rules")
 	if err != nil {
 		log.Fatal(err)
 	}
+	FS = wrapFS{rulesFS.(subFS)}
+}
+
+// wrapFS is a wrapper around a subFS that removes the prefix from the
```

Comment:
I don't quite understand what this wrapper is doing - perhaps some unit tests for it can help clarify. Notably, I don't understand what "The rationale is that the coraza parser
// will try to resolve an absolute path seeking for the trailing /" means and why removing all content up to a `@` could help, seems like a cludge?

Reply:
Makes total sense. I will come up with a test for this.